### PR TITLE
fix(chat): fix conversation selection after deleting all conversations (Issue #2056)

### PR DESCRIPTION
**Pull request by Alexander-Kezik**
**Description:**

fix conversation selection after deleting all conversations

Issues:

- Issue #2056

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2626.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat/src/utils/app/publications.ts [hunk 1]
```diff
@@ -142,27 +142,32 @@ export const mapPublishedItems = <T extends PromptInfo | ConversationInfo>(
...
-        id: itemId,
+        id: item.id,
         folderId,
         publishedWithMe: isRootId(folderId),
-      } as T);
+      } as T;
+
+      if (featureType === FeatureType.Chat) {
+        (itemToAdd as ConversationInfo).lastActivityDate = item.updatedAt;
+      }
```
Comment (IlyaBondar) [inline apps/chat/src/utils/app/publications.ts]: do we need this condition?

File: apps/chat/src/store/conversations/conversations.epics.ts [hunk 2]
```diff
@@ -2431,23 +2431,24 @@ const uploadFoldersEpic: AppEpic = (action$, state$) =>
...
+            (folder) => folder.folders,
+          );
+          const conversations = foldersAndEntities.flatMap(
+            (items) => items.entities,
+          );
+          const publicConversations = conversations
+            .filter((conv) => isEntityIdPublic(conv))
+            .map((conv) => ({
+              ...conv,
+              updatedAt: conv.lastActivityDate ?? Date.now(),
```
Comment (IlyaBondar) [inline apps/chat/src/store/conversations/conversations.epics.ts]: Why not `lastActivityDate`?
```suggestion
              lastActivityDate: conv.lastActivityDate ?? Date.now(),
```

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2626.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2626.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2626.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by IlyaBondar — APPROVED** [review]

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2626.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2626.nightly-test.deltixhub.io
>E2E tests status: success

**Pull request merged**